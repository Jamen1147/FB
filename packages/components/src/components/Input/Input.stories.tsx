import Input from './Input';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'Input',
  component: Input,
} as Meta;

const Template: Story<React.ComponentProps<typeof Input>> = (args) => (
  <Input {...args} />
);

const args = {
  placeholder: 'Input',
};

export const Outlined = Template.bind({});
Outlined.args = {
  ...args,
  variant: 'outlined',
  onChange: () => console.log('changing'),
};

export const Filled = Template.bind({});
Filled.args = {
  ...args,
  variant: 'filled',
};
