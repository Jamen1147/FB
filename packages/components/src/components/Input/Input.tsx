import { styled, TextField } from '@mui/material';
import React from 'react';

const MUIInput = styled(TextField)`
  input {
    padding: ${({ theme }) => theme.spacing(1)};
  }
`;

const Input = (props: React.ComponentProps<typeof TextField>) => (
  <MUIInput {...props} />
);

export default Input;
