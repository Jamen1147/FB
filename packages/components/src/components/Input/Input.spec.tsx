import Input from './Input';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('<Input />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Input label="label" placeholder="input" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onChange when type', () => {
    const onChange = jest.fn();
    render(<Input label="label" placeholder="input" onChange={onChange} />);
    const input = screen.getByRole('textbox');
    userEvent.type(input, 'something');
    expect(onChange).toHaveBeenCalled();
  });
});
