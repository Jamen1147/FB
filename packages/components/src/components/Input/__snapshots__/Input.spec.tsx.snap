// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-ev9kko-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="mui-1"
      id="mui-1-label"
    >
      label
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="mui-1"
        placeholder="input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-173wfxe"
        >
          <span>
            label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
